File: Data Structures & Algorithms/boats-to-save-people/submission-2.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& nums, int limit) 
    {
       sort(nums.begin(),nums.end());
       int left=0,right=nums.size()-1,count=0;
       while(left<=right)
       {
        if(nums[left]+nums[right]<=limit)
        {
            count++;
            left++;
            right--;
        }
        else if(nums[right]<=limit)
        {
            count++;
            right--;
        }
        else if(nums[left]<=limit)
        {
            count++;
            left++;
        }
       }   
       
       return count;
        
    }
};

File: Data Structures & Algorithms/minimum-number-of-operations-to-make-array-empty/submission-2.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        mp={}
        for i in nums:
            mp[i]=mp.get(i,0)+1

        count=0
        for key,value in mp.items():
             if(value==1):
                return -1
             else:
                count+=(value+2)//3;

        return count    



        